clang/test/SemaSYCL/scheduler_target_fmax_mhz.cpp

```diff
@@ -0,0 +1,36 @@
+// RUN: %clang_cc1 %s -fsyntax-only -fsycl -fsycl-is-device -triple spir64 -Wno-sycl-2017-compat -verify
+// RUN: %clang_cc1 %s -fsyntax-only -fsycl -fsycl-is-device -triple spir64 -DTRIGGER_ERROR -Wno-sycl-2017-compat -verify
+// RUN: %clang_cc1 %s -fsyntax-only -ast-dump -fsycl -fsycl-is-device -triple spir64 -Wno-sycl-2017-compat | FileCheck %s
+
+#include "Inputs/sycl.hpp"
+#ifndef TRIGGER_ERROR
+[[intelfpga::scheduler_target_fmax_mhz(2)]] // expected-no-diagnostics
+void
+func() {}
+#endif // TRIGGER_ERROR
+
+int main() {
+#ifndef TRIGGER_ERROR
+  // CHECK-LABEL:  -FunctionDecl {{.*}}test_kernel1 'void ()'
+  // CHECK:        SYCLIntelSchedulerTargetFmaxMhzAttr {{.*}} 5
+  cl::sycl::kernel_single_task<class test_kernel1>(
+      []() [[intelfpga::scheduler_target_fmax_mhz(5)]]{});
+
+  // CHECK-LABEL:  `-FunctionDecl {{.*}}test_kernel2 'void ()'
+  // CHECK:  -SYCLIntelSchedulerTargetFmaxMhzAttr {{.*}} 2
+  cl::sycl::kernel_single_task<class test_kernel2>(
+      []() { func(); });
+
+#else
+  [[intelfpga::scheduler_target_fmax_mhz(0)]] int Var = 0; // expected-error{{'scheduler_target_fmax_mhz' attribute only applies to functions}}
```

Comment:
Add some examples where the value is dependent (that is, result of a template parameter).